app/plugins/Files/js/buttons.js

```diff
@@ -177,13 +177,24 @@ eID('upload-dir').onclick = function() {
 	uploadDir(loadPath, nickname);
 };
 
-/*
-eID('search-bar').keydown(function(e) {
-	// TODO: figure out search bar
-	updateFileList(lastLoadedFiles);
-};
-eID('search').onclick = function(e) {
-	// TODO: figure out search bar
-	updateFileList(lastLoadedFiles);
-};
-*/
+// Start search when typing in Search field
+eID('search-bar').onkeyup = function() {
+	tooltip('Searching...', this);
+	var searchstr = eID('search-bar').value;
+	filterFileList(searchstr);
+};
+
+// Filter file list by search string
+function filterFileList(searchstr) {
+	NodeList.prototype.forEach = Array.prototype.forEach
```

Comment:
This is Javascript, that's legal. ;)

Automatic Semicolon Insertion (ASI) can be wicked though in certain cases resulting in obtuse behavior  and is best avoided.

http://bclary.com/2004/11/07/#a-7.9.1